python/test_infra/test_utils.py

```diff
@@ -204,86 +190,136 @@ def ttir_to_ttmetal(
 
-    def decorator(fn: Callable):
-        def wrapper(*args, **kwargs):
-            # First, call the decorated function to get the MLIR module.
-            module, builder = fn(*args, **kwargs)
+    # Now, pass it through the TTIR to TTMetal pipeline. Module gets
+    # modified in place.
+    ttir_to_ttmetal_backend_pipeline(module, f"system-desc-path={system_desc_path}")
 
-            # Now, pass it through the TTIR to TTMetal pipeline. Module gets
-            # modified in place.
-            ttir_to_ttmetal_backend_pipeline(
-                module, f"system-desc-path={system_desc_path}"
-            )
+    print("`ttir_to_ttmetal_backend_pipeline` passed successfully.")
 
-            print("`ttir_to_ttmetal_backend_pipeline` passed successfully.")
+    # Optionally dump to file.
+    if dump_to_file:
+        with open(output_file_name, "w") as f:
+            f.write(str(module))
 
-            # Optionally dump to file.
-            if dump_to_file:
-                with open(output_file_name, "w") as f:
-                    f.write(str(module))
-
-            return module, builder
-
-        return wrapper
-
-    return decorator
+    return module
 
 
 def ttnn_to_flatbuffer(
+    module,
+    builder,
     output_file_name: str = "ttnn_fb.ttnn",
 ):
     """
-    Converts TTNN module to flatbuffer and saves to file, meant to be used as a
-    decorator on top of `ttir_to_ttnn` decorator. Take note that `ttir_to_ttnn`
-    has to return module instead of file name if decorated with this decorator.
-
-    Wrapper around `ttnn_to_flatbuffer_file` pybound pass.
+    Converts TTNN module to flatbuffer and saves to file. Wrapper around
+    `ttnn_to_flatbuffer_file` pybound pass.
     """
 
-    def decorator(test_fn: Callable):
-        def wrapper(*args, **kwargs):
-            # Get the TTNN module by calling the wrapped function.
-            module, builder = test_fn(*args, **kwargs)
+    # Convert to flatbuffer file.
+    ttnn_to_flatbuffer_file(module, output_file_name, builder.get_golden_map())
 
-            # Convert to flatbuffer file.
-            ttnn_to_flatbuffer_file(module, output_file_name, builder.get_golden_map())
+    print("`ttnn_to_flatbuffer_file` passed successfully.")
 
-            print("`ttnn_to_flatbuffer_file` passed successfully.")
 
-        return wrapper
+def ttmetal_to_flatbuffer(
+    module,
+    builder,
+    output_file_name: str = "ttmetal_fb.ttm",
+):
+    """
+    Converts TTMetal module to flatbuffer and saves to file. Wrapper around
+    `ttmetal_to_flatbuffer_file` pybound pass.
+    """
 
-    return decorator
+    # Convert to flatbuffer file.
+    ttmetal_to_flatbuffer_file(module, output_file_name, builder.get_golden_map())
 
+    print("`ttmetal_to_flatbuffer_file` passed successfully.")
 
-def ttmetal_to_flatbuffer(
-    output_file_name: str = "ttmetal_fb.ttmg",
+
+# ----- Decorators for doing passes and compiling to flatbuffer -----
+
+
+def compile_and_convert(
```

Comment:
Maybe we can call this just `compile`?

Reply:
That seems a little generic to me, especially if we plan to integrate this into the rest of the LIT tests. I could be wrong here though, LMK what you think.

Reply:
How about `compile_to_flatbuffer`?

Reply:
Oooo I like that much better